koku/api/provider/provider_manager.py

```diff
@@ -107,6 +107,51 @@ def get_previous_month_data_exists(self):
             completed_datetime__isnull=False,
         ).exists()
 
+    def get_downloading_state(self):
+        """Get latest manifest downloading state."""
+        state = "pending"
+        manifest = CostUsageReportManifest.objects.filter(
+            provider=self._uuid,
+            billing_period_start_datetime=self.date_helper.this_month_start,
+        ).first()
+        if manifest:
+            if manifest.state.get("download"):
+                if manifest.state.get("download").get("start"):
+                    state = "in-progress"
+                if manifest.state.get("download").get("end"):
+                    state = "complete"
+        return state
+
+    def get_processing_state(self):
+        """Get latest manifest processing state."""
+        state = "pending"
+        manifest = CostUsageReportManifest.objects.filter(
+            provider=self._uuid,
+            billing_period_start_datetime=self.date_helper.this_month_start,
+        ).first()
+        if manifest:
+            if manifest.state.get("processing"):
+                if manifest.state.get("processing").get("start"):
+                    state = "in-progress"
+                if manifest.state.get("processing").get("end"):
+                    state = "complete"
+        return state
+
+    def get_summary_state(self):
+        """Get latest manifest summary state."""
+        state = "pending"
+        manifest = CostUsageReportManifest.objects.filter(
+            provider=self._uuid,
+            billing_period_start_datetime=self.date_helper.this_month_start,
+        ).first()
+        if manifest:
+            if manifest.state.get("summary"):
+                if manifest.state.get("summary").get("start"):
+                    state = "in-progress"
+                if manifest.state.get("summary").get("end"):
+                    state = "complete"
+        return state
```

Comment:
I worry about the extra DB calls we are doing here. All 3 of these do the exact same DB query. Could we make this query a cached property or something? Or make this a single function that returns the info we need in a dict? Or maybe something else?

I'm also curious why the Manifest itself doesn't carry the pending, in-progress, and complete states itself...